src/main/java/games/strategy/engine/lobby/server/db/UserController.java

```diff
@@ -175,10 +212,18 @@ public DBUser getUserByName(final String userName) {
       ps.close();
       return user;
     } catch (final SQLException sqle) {
-      s_logger.info("Error for testing user existence:" + userName + " error:" + sqle.getMessage());
+      logger.info("Error for testing user existence:" + userName + " error:" + sqle.getMessage());
       throw new IllegalStateException(sqle.getMessage());
     } finally {
       DbUtil.closeConnection(con);
     }
   }
+
+  private String mask(final HashedPassword hashedPassword) {
+    return mask(hashedPassword.value);
+  }
+
+  private String mask(final String hashedPassword) {
+    return hashedPassword.replaceAll(".", "*");
```

Comment:
We are disclosing the length of `hashedPassword` here.  Is that valuable information that can be used by an attacker if it was leaked?  I haven't followed all the recent changes, but is `hashedPassword` always either a SHA-512 hash or an MD5-crypt-encoded hash/salt?  If so, then I guess that's ok as each string will always have the same length.  I'm assuming you're doing that (as opposed to just not logging anything about the password) so you can tell if it's a password hashed by the old system or new system based on its length?

Nit: the `mask()` methods look like they can be made static.